src/ashley/api/views.py

```diff
@@ -0,0 +1,106 @@
+"""Views of the ashley api django application."""
+import logging
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from django.core.exceptions import PermissionDenied
+from machina.apps.forum_permission.viewmixins import (
+    PermissionRequiredMixin as BasePermissionRequiredMixin,
+)
+from machina.core.db.models import get_model
+from machina.core.loading import get_class
+from rest_framework import mixins, status, viewsets
+from rest_framework.response import Response
+
+from ashley.context_mixins import get_current_lti_session
+from ashley.defaults import _FORUM_ROLE_MODERATOR, _FORUM_ROLE_STUDENT
+from ashley.permissions import ManageModeratorPermission
+
+from .serializers import UserSerializer
+
+User = get_user_model()
+LTIContext = get_model("ashley", "LTIContext")
+Forum = get_model("forum", "Forum")
+PermissionRequiredMixin: BasePermissionRequiredMixin = get_class(
+    "forum_permission.viewmixins", "PermissionRequiredMixin"
+)
+logger = logging.getLogger(__name__)
+
+
+class UserModeratorListApiView(mixins.ListModelMixin, viewsets.GenericViewSet):
+    """API requests to list moderators for current LTIContext"""
+
+    # Add permission to check if user has the right permission and can manage moderators
+    permission_classes = [ManageModeratorPermission]
+    serializer_class = UserSerializer
+
+    def get_queryset(self):
+        """List all the users that are moderators. Moderators have the primitive role
+        student."""
+        context = get_current_lti_session(self.request)
+        return User.objects.filter(
+            is_active=True,
+            groups__name=context.get_group_role_name(_FORUM_ROLE_MODERATOR),
+        ).filter(groups__name=context.get_group_role_name(_FORUM_ROLE_STUDENT))
+
+
+class UserStudentListApiView(mixins.ListModelMixin, viewsets.GenericViewSet):
+    """API requests to list students for current LTIContext and to update students
+    to promote or revoke moderators."""
+
+    # Add permission to check if user can manage moderator
+    permission_classes = [ManageModeratorPermission]
+    serializer_class = UserSerializer
+
+    def get_queryset(self):
+        "Gets all the students for the current LTIContext."
+        context = get_current_lti_session(self.request)
+        return User.objects.filter(
+            is_active=True,
+            groups__name=context.get_group_role_name(_FORUM_ROLE_STUDENT),
+        ).exclude(groups__name=context.get_group_role_name(_FORUM_ROLE_MODERATOR))
+
+    def update(self, request):
```

Comment:
What is made in this method is wrong IMO.
It starts with the url patterns used and how you deal with it.
First of all the route should be a `PATCH` on  `/users/{user_id}` and in the body you should have an entire user object. So the role should also be in the serializer (I made a comment about that).

I also thinks that having 2 classes managing a user but with different roles is wrong.

classes `UserModeratorListApiView` and `UserStudentListApiView` share the same permission class, they return the same kind of list (a user). So I think you should only have one class `UserApiView`, in the list function you can filter the user you want by adding a querystring `role`. Without this querystring you return all the users related to this forum, if you want a specific role your should filter with the querystring like this `?role=[moderator,student]`, I wouldn't be surprise DRF can handle this case.